test/pxd_test.cc

```diff
@@ -0,0 +1,396 @@
+#include <stdlib.h>
+#include <gtest/gtest.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <linux/fuse.h>
+#include <sys/poll.h>
+#include <boost/iostreams/device/file_descriptor.hpp>
+#include <sys/uio.h>
+#include <string>
+#include <boost/lexical_cast.hpp>
+#include <functional>
+#include <linux/fs.h>
+#include <sys/ioctl.h>
+#include <thread>
+#include <vector>
+#include "pxd.h"
+#include "fuse.h"
+
+using namespace std::placeholders;
+
+std::string control_device(unsigned int driver_context_id)
+{
+	assert(driver_context_id < PXD_NUM_CONTEXTS);
+	std::string ret{PXD_CONTROL_DEV};
+	if (driver_context_id != 0)
+		ret += "-" + std::to_string(driver_context_id);
+	return ret;
+}
+
+class PxdTest : public ::testing::Test {
+protected:
+	int fd;		// control file descriptor
```

Comment:
would it be acceptable to rename this to ctl_fd.. test deals with too many file handles, so it helps to be more specific.

Reply:
renamed